### ISSUE TITLE
Another issue with nested any { it.all { } }

### ISSUE THREAD

**Issue by bschelberg** [willowtreeapps/assertk#354]
Following on from #331 (fixed now, thanks), another situation I've come across where the nested any, all doesn't behave as expected. Please check out my updated project: https://github.com/bschelberg/assertk-test-issue. Again there is just one test that should fail but it doesn't. Curiously, removing one of the pairs in the list results in the test failing as expected.

Test code:

    @Test
    fun `should not pass, pairs`() {
        val key1 = "key 1"
        val value1 = "value 1"

        val pairs = listOf("key 2" to "value 2", "key 2" to "value 2", "key 2" to "value 2", key1 to value1)

        assertThat(pairs).any {
            it.all {
                transform { it.first }.isEqualTo(key1)
                transform { it.second }.isEqualTo("unexpected")
            }
        }
    }

**Comment by evant** [comment]
Yep, good catch! Looks like there was a bug in the any impl that broke if you had enough items in the list 🤦 .
